tools/printast/printast.ml

```diff
@@ -0,0 +1,26 @@
+open Parser_extended
+
+let get_arg () =
+  if Array.length Sys.argv < 2 then (
+    Printf.eprintf "Not enough argument\n" ;
+    exit 2 )
+  else Sys.argv.(1)
+
+let parse_and_print parsef printf lexbuf =
+  printf Format.std_formatter (parsef lexbuf)
+
+let get_ppf fname =
+  match Filename.extension fname with
+  | ".mli" ->
+      (parse_and_print Parse.interface Printast.interface, "interface")
+  | _ ->
+      ( parse_and_print Parse.implementation Printast.implementation
+      , "implementation" )
```

Comment:
Maybe `Bin_conf.kind_of_ext` could be re-used (moved to `Syntax` maybe? since it returns a `Syntax.t`)

Reply:
Good idea! I also moved the match on the syntax to `Extended_ast`.